libraries/core-react/src/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,208 @@
+  justify-content: space-between;
+  flex-wrap: nowrap;
+  align-items: center;
+`
+
+const Text = styled(Typography)`
+  white-space: nowrap;
+`
+
+function getAriaLabel(page, selected) {
+  return `${selected === page ? 'Current page, ' : 'Go to '}page ${page}`
+}
+
+export const Pagination = forwardRef(function Pagination(
+  {
+    totalItems,
+    defaultPage,
+    withItemIndicator,
+    itemsPerPage,
+    className,
+    ...other
+  },
+  ref,
+) {
+  const pages = Math.ceil(totalItems / itemsPerPage) // Total page numbers
+  const columns = pages < 5 ? pages + 2 : 7 // Total pages to display on the control + 2:  < and >
+
+  const [activePage, setActivePage] = useState(defaultPage)
+
+  const currentItemFirst =
+    activePage === 1 ? 1 : activePage * itemsPerPage - itemsPerPage + 1 // First number of range of items at current page
+  const currentItemLast =
+    activePage === pages ? totalItems : activePage * itemsPerPage // Last number of range of items at current page
+
+  const handleClick = (page) => () => {
+    setActivePage(page)
+  }
+
+  const moveLeft = () => {
+    if (activePage > 1) {
+      setActivePage(activePage - 1)
+    }
+  }
+
+  const moveRight = () => {
+    if (activePage < pages) {
+      setActivePage(activePage + 1)
+    }
+  }
+
+  const items = PaginationControl(pages, activePage)
+
+  const props = {
+    ref,
+    withItemIndicator,
+    className,
+    ...other,
+  }
+
+  const pagination = (
+    <Navigation aria-label="pagination" {...props}>
+      <UnorderedList
+        style={{
+          gridTemplateColumns: 'repeat(' + columns + ', 48px)',
+        }}
+      >
+        <Button
+          variant="ghost_icon"
+          onClick={moveLeft}
+          disabled={activePage === 1}
+          aria-label="Go to previous page"
+        >
+          <Icon name="chevron_left" title="previous" />
+        </Button>
+        {items.length > 0 &&
+          items.map((page, index) =>
+            page !== 'ELLIPSIS' ? (
+              // eslint-disable-next-line react/no-array-index-key
+              <ListItem key={'list-item' + index}>
+                <PaginationItem
+                  {...page}
+                  aria-label={getAriaLabel(page, activePage)}
+                  aria-current={activePage}
+                  page={page}
+                  selected={page === activePage}
+                  onClick={handleClick(page)}
+                />
+              </ListItem>
+            ) : (
+              <StyledIcon
+                // eslint-disable-next-line react/no-array-index-key
+                key={'ellipsis-' + index}
+                name="more_horizontal"
+                title="ellipsis"
+              />
+            ),
+          )}
+        <Button
+          variant="ghost_icon"
+          onClick={moveRight}
+          aria-label="Go to next page"
+          disabled={activePage === pages}
+        >
+          <Icon name="chevron_right" title="next" />
+        </Button>
```

Comment:
A list can't have buttons and icons (ellipsis) as direct children :) Should wrap them in li